src/goose.rs

```diff
@@ -1224,20 +1221,22 @@ impl GooseClient {
         GooseResponse::new(raw_request, response)
     }
 
+    /// Helper that pushes statistics to parent thread, when defined.
     fn send_to_parent(&self, raw_request: &GooseRawRequest) {
-        let parent = match self.parent.clone() {
-            Some(p) => p,
-            None => {
-                error!("unable to communicate with parent thread, exiting");
-                std::process::exit(1);
-            }
-        };
-        match parent.send(raw_request.clone()) {
-            Ok(_) => (),
-            Err(e) => {
-                error!("unable to communicate with parent thread, exiting: {}", e);
-                std::process::exit(1);
+        match self.parent.clone() {
+            Some(p) => {
+                let parent = p;
+                match parent.send(raw_request.clone()) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        info!("unable to communicate with parent thread, exiting: {}", e);
+                        std::process::exit(1);
+                    }
+                }
             }
+            // No parent thread is defined when running test_start_task,
+            // test_stop_task, and during testing.
+            None => (),
```

Comment:
nit - non test related changes should be separate PR (and justified also without the test)

Reply:
This change was required for tests to work (granted it _also_ meant that `test_start_task` and `test_stop_task` wasn't working).